libvita2d/source/vita2d_image_jpeg.c

```diff
@@ -6,23 +6,14 @@
 #include <jpeglib.h>
 #include "vita2d.h"
 
+// Max memory we want to use by jpg is 64 MB half of the VRAM, how every pixel is 4 bytes, then 64 * 1024 * 1024 / 4
+// Which is exactly the same than a picture of 4096 x 4096
+#define MAX_JPEG_MATRIX_SIZE 16777216
```

Comment:
To be honest, I don't understand where this "Max memory we want to use by jpg is 64 MB half of the VRAM" requirement comes from.
The previous code didn't have such requirements.

Reply:
Yes, you have it, you have it as `4096` width and height.
With your preivous code, an image with size, `8000x 50` will be reduced to `4000x25`
However, with my approach, you still can load `800x50` image

Reply:
Isn't 4096 the limit for both width and height? I mean that even if only width, or only height is bigger than 4096, then the texture is not supported by the HW.

Reply:
`sceGxmTextureInitLinear` wants `width` to be between 1 and 4096, and the same for `height`.

Reply:
I didn't know that the width of the texture can not be bigger than `4096` width/height, can we double-check it?

In this case we can do something like this
```
float downScale = (float)MAX(jinfo->image_width, jinfo->image_height) / 4096;
jinfo->scale_denom = ceil(downScale);
```